packages/ckeditor5-engine/src/view/downcastwriter.js

```diff
@@ -685,43 +695,46 @@ export default class DowncastWriter {
 		// Check if nodes to insert are instances of AttributeElements, ContainerElements, EmptyElements, UIElements or Text.
 		validateNodesToInsert( nodes, this.document );
 
-		const container = getParentContainer( position );
+		// Group nodes in batches of nodes that require or do not require breaking an AttributeElements.
+		const nodeGroups = nodes.reduce( ( groups, node ) => {
+			const lastGroup = groups[ groups.length - 1 ];
 
-		if ( !container ) {
-			/**
-			 * Position's parent container cannot be found.
-			 *
-			 * @error view-writer-invalid-position-container
-			 */
-			throw new CKEditorError(
-				'view-writer-invalid-position-container',
-				this.document
-			);
-		}
+			// Break attributes on nodes that do exist in the model tree so they can have attributes, other elements
+			// can't have an attribute in model and won't get wrapped with an AttributeElement while down-casted.
+			const breakAttributes = !node.is( 'uiElement' );
```

Comment:
I'm not sure if this should not include `rawElement` also.

Reply:
But OTOH we could use `RawElement` (instead of `UIElement`) for suggestions in TC.